kyo-direct/shared/src/test/scala/kyo/CoreTest.scala

```diff
@@ -0,0 +1,195 @@
+package kyo
+
+class CoreTest extends Test:
+
+    "atomic operations" - {
+        "AtomicInt" in run {
+            defer {
+                val counter = AtomicInt.init(0).now
+                counter.incrementAndGet.now
+                counter.incrementAndGet.now
+                counter.decrementAndGet.now
+                assert(counter.get.now == 1)
+            }
+        }
+
+        "AtomicRef" in run {
+            defer {
+                val ref = AtomicRef.init("initial").now
+                ref.set("updated").now
+                assert(ref.get.now == "updated")
+            }
+        }
+    }
+
+    "clock operations" - {
+        "sleep and timeout" in run {
+            defer {
+                val start = Clock.now.now
+                Async.sleep(5.millis).now
+                val elapsed = Clock.now.now - start
+                assert(elapsed >= 5.millis)
+            }
+        }
+
+        "deadline" in run {
+            defer {
+                val deadline = Clock.deadline(1.second).now
+                assert(!deadline.isOverdue.now)
+                assert(deadline.timeLeft.now <= 1.second)
+            }
+        }
+    }
+
+    // TODO Compiler crash because `Queue` is an opaque type without a type bound
```

Comment:
I can't use `pendingUntilFixed` because the compiler crashes. I'll create an issue for this